libs/sqlite3/Makefile

```diff
@@ -8,19 +8,23 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=sqlite
-PKG_VERSION:=3470200
+PKG_VERSION:=3.49.1
 PKG_RELEASE:=1
 
-PKG_SOURCE:=$(PKG_NAME)-autoconf-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=https://www.sqlite.org/2024/
-PKG_HASH:=f1b2ee412c28d7472bc95ba996368d6f0cdcf00362affdadb27ed286c179540b
+PadPart=$(if $(shell [ '$(1)' -lt 10 ] && echo 1),0$(1),$(1))
```

Comment:
Perhaps `$(shell printf "%02d" $(1))`? ( a bit easier to grok )

Reply:
Thanks. Much better indeed. I don't know what I was thinking :sweat_smile:

Reply:
> this has the consequence of the new version being less than the previous version, so it will not show as an update for the existing installations

IMO this makes simple things complex, not seeing much benefit from this.

Reply:
I thought it was kind of off that the version didn't match the upstream and was specified in a non-semver-ish looking manner.

Reply:
> IMO this makes simple things complex, not seeing much benefit from this.

&lt;irony&gt;To be fair, the prevailing wisdom for a long time was never to update your db server (within reason). &lt;/irony&gt;

I see no harm in going this route (changing to semver) since updates to this are infrequent. And owut and sysupgrade replace the whole system conveniently anyway.

I'm on the fence on this one (I had to stare at the code for a minute to figure it out), though I'm fine with either way.

Reply:
I checked some other distros for references: they all use the _real_ version, by hardcoding both formats or vendoring the source outside of the specs with the _real_ version. Here's the relevant parts of the latest spec from Fedora:

```spec
%define majorver 3
%define realver 3490100
%define docver 3490100
%define rpmver 3.49.1
%define year 2025

...

Name: sqlite
Version: %{rpmver}

...

Source0: http://www.sqlite.org/%{year}/sqlite-src-%{realver}.zip
Source1: http://www.sqlite.org/%{year}/sqlite-doc-%{docver}.zip
Source2: http://www.sqlite.org/%{year}/sqlite-autoconf-%{realver}.tar.gz
```

I can remove the conversion bits and just have two versions instead. What do you think?

Reply:
> I see no harm in going this route (changing to semver) since updates to this are infrequent. And owut and sysupgrade replace the whole system conveniently anyway.

It's hard to say whether upgrading packages or upgrading the system is more common.
I'm a bit against this change.

> \<irony\>To be fair, the prevailing wisdom for a long time was never to update your db server (within reason). \</irony\>

<img src="https://i.programmerhumor.io/2024/04/programmerhumor-io-programming-memes-8b0dd05bebe8249.jpg" />

Reply:
We should just leave it out of the current stable, i.e. update this package using the existing versioning there. And it won't matter for new installs. Go for example has already diverged in stable.